src/input.rs

```diff
@@ -276,14 +276,26 @@ impl<'a> InputReader<'a> {
             return Ok(true);
         }
 
-        let res = self.inner.read_until(b'\n', buf).map(|size| size > 0)?;
+        let res = Self::read_line_safely(&mut self.inner, buf).map(|size| size > 0)?;
 
         if self.content_type == Some(ContentType::UTF_16LE) {
             let _ = self.inner.read_until(0x00, buf);
         }
 
         Ok(res)
     }
+
+    fn read_line_safely<R: BufRead + 'a>(reader: &mut R, buf: &mut Vec<u8>) -> io::Result<usize> {
+        let limit = 100_000_000_usize;
+
+        match reader.take(limit as u64).read_until(b'\n', buf) {
```

Comment:
Note: `take` does not actually take anything, so the name is a bit odd. It just puts an upper limit. See https://doc.rust-lang.org/std/io/trait.Read.html#method.take